service/awsconfig/v10/resource/cloudformation/delete.go

```diff
@@ -4,26 +4,35 @@ import (
 	"context"
 
 	"github.com/aws/aws-sdk-go/aws"
-	awscloudformation "github.com/aws/aws-sdk-go/service/cloudformation"
-	"github.com/giantswarm/aws-operator/service/awsconfig/v10/key"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/framework"
+
+	"github.com/giantswarm/aws-operator/service/awsconfig/v10/key"
 )
 
 func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange interface{}) error {
 	customObject, err := key.ToCustomObject(obj)
 	if err != nil {
 		return microerror.Mask(err)
 	}
-	stackInputToDelete, err := toStackState(deleteChange)
+	stackStateToDelete, err := toStackState(deleteChange)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
-	if stackInputToDelete.Name != "" {
+	if stackStateToDelete.Status == cloudformation.ResourceStatusUpdateInProgress {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "cannot delete CF stacks due to stack state transitioning")
+
+		// TODO control flow via more proper mechanism via something like the
+		// context like it is done for cancelation already.
+		return microerror.Maskf(deletionMustBeRetriedError, "stack state is transitioning")
+	}
```

Comment:
In case an update is happening we go for the next resync. CF is not able to overwrite actions being processed already. So we have to wait and try again. One step at a time.